subsys/testsuite/ztest/src/ztest_new.c

```diff
@@ -605,11 +626,132 @@ void end_report(void)
 K_APPMEM_PARTITION_DEFINE(ztest_mem_partition);
 #endif
 
+static void __ztest_init_unit_test_result_for_suite(struct ztest_suite_node *suite)
+{
+	struct ztest_unit_test *test = NULL;
+
+	while (((test = z_ztest_get_next_test(suite->name, test)) != NULL)) {
+		test->stats->run_count = 0;
+		test->stats->skip_count = 0;
+		test->stats->fail_count = 0;
+		test->stats->pass_count = 0;
+	}
+}
+
+static void flush_log(void)
+{
+	if (IS_ENABLED(CONFIG_LOG_PROCESS_THREAD)) {
+		while (log_data_pending()) {
+			k_sleep(K_MSEC(10));
+		}
+		k_sleep(K_MSEC(10));
+	} else {
+		while (LOG_PROCESS()) {
+		}
+	}
+}
+
+/* Show one line summary for a test suite.
+ */
+static void __ztest_show_suite_summary_oneline(struct ztest_suite_node *suite)
+{
+	int distinct_pass = 0, distinct_fail = 0, distinct_skip = 0, distinct_total = 0;
+	int passrate = 0, passrate1 = 0, passrate2 = 0;
+	int suite_result = TC_PASS;
+
+	struct ztest_unit_test *test = NULL;
+
+	/** summary of disctinct run  */
+	while (((test = z_ztest_get_next_test(suite->name, test)) != NULL)) {
+		distinct_total++;
+		if (test->stats->skip_count == test->stats->run_count) {
+			distinct_skip++;
+		} else if (test->stats->pass_count == test->stats->run_count) {
+			distinct_pass++;
+		} else {
+			distinct_fail++;
+		}
+	}
+
+	if (distinct_skip == distinct_total) {
+		suite_result = TC_SKIP;
+		passrate1 = passrate2 = 0;
+	} else {
+		suite_result = (distinct_fail > 0) ? TC_FAIL : TC_PASS;
+		passrate = distinct_pass * 100;
+		passrate1 = passrate / (distinct_total - distinct_skip);
+		passrate2 = (passrate - passrate1 * (distinct_total - distinct_skip)) / 100;
+	}
+
+	TC_SUMMARY_PRINT("SUITE %s - %3d.%02d%% [%s]: pass = %d, fail = %d, "
+				"skip = %d, total = %d\n",
+				TC_RESULT_TO_STR(suite_result),
+				passrate1, passrate2,
+				suite->name, distinct_pass, distinct_fail,
+				distinct_skip, distinct_total);
+	flush_log();
+}
+
+#ifdef CONFIG_ZTEST_VERBOSE_SUMMARY
+static void __ztest_show_suite_summary_verbose(struct ztest_suite_node *suite)
+{
+	struct ztest_unit_test *test = NULL;
+	int tc_result = TC_PASS;
+	int flush_frequency = 0;
+
+	__ztest_show_suite_summary_oneline(suite);
+
+	while (((test = z_ztest_get_next_test(suite->name, test)) != NULL)) {
+		if (test->stats->skip_count == test->stats->run_count) {
+			tc_result = TC_SKIP;
+		} else if (test->stats->pass_count == test->stats->run_count) {
+			tc_result = TC_PASS;
+		} else {
+			tc_result = TC_FAIL;
+		}
+
+		TC_SUMMARY_PRINT(" - %s - [%s.%s]\n", TC_RESULT_TO_STR(tc_result),
+				 test->test_suite_name, test->name);
+
+		if (flush_frequency % 2 == 0) {
```

Comment:
Won't this code just be run once?

`flush_frequency` starts out at `0` and then we increment it to `1` but then never change it so it stays at `1` and this branch only taken once ever since `(1 % 1) == 0` is a false predicate.

Reply:
But I do have a `flush_frequency++` at line 734 and it is `%2`.

Reply:
Yes, you are right. The `flush_frequency++` should be outside of the loop.